tests/test_proxy_use.py

```diff
@@ -78,25 +75,21 @@ def setUpClass(cls):
                                 " (proxy_server.py is Python2 only)")
 
     # Launch a simple HTTP server (serves files in the current dir).
-    cls.http_port = random.randint(30000, 45000)
-    cls.http_server_proc = popen_python(
-        ['simple_server.py', str(cls.http_port)])
+    cls.http_server_handler = utils.TestServerProcess(log=logger)
 
     # Launch an HTTPS server (serves files in the current dir).
-    cls.https_port = cls.http_port + 1
-    cls.https_server_proc = popen_python(
-        ['simple_https_server.py', str(cls.https_port)])
-
+    cls.https_server_handler = utils.TestServerProcess(log=logger,
+        server='simple_https_server.py')
```

Comment:
This has the same issue as `test_https_connection()` -- there is now a (very small) chance of collision with cls.http_server_handler.port.

I think the real solution is to make simple_server actually report back that it did succeed in binding the port (so TestServerProcess can retry if it did not). It should probably (?) not be part of this PR though: you'll have to choose an interim solution here: I think I'd use `port=cls.http_server_handler.port + 1` because that is closest to current functionality

Reply:
this applies to all servers below as well

Reply:
Yes, I won't fix this in this pr, but I will return it to the previous state in both `test_proxy_use.py` and `test_download.py`.